### PR TITLE
[5.1] Allow fetch a specific value from the Route action array

### PR DESCRIPTION
**Pull request by ibrasho**
Solves: https://github.com/laravel/framework/issues/9153

### CODE REVIEW

File: src/Illuminate/Routing/Route.php [hunk 1]
```diff
@@ -905,13 +905,13 @@ public function getActionName()
     }
 
     /**
-     * Get the action array for the route.
+     * Get the action array, or a specific key, for the route.
      *
      * @return array
      */
-    public function getAction()
+    public function getAction($key = null)
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: You need to update the phpdoc.

**Comment by GrahamCampbell** [comment]
Please send this to 5.2 as it's a breaking change.

**Comment by GrahamCampbell** [comment]
Actually, looks good. :P

File: src/Illuminate/Routing/Route.php [hunk 2]
```diff
@@ -905,13 +905,13 @@ public function getActionName()
     }
 
     /**
-     * Get the action array for the route.
+     * Get the action array, or a specific key, for the route.
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: Reads better without the commas.

Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: And without the "a".

File: src/Illuminate/Routing/Route.php [hunk 3]
```diff
@@ -905,13 +905,13 @@ public function getActionName()
...
      *
      * @return array
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: And this is now wrong too. Should be `string|array`.

Comment (ibrasho) [inline src/Illuminate/Routing/Route.php]: Why not `array|mixed`?

Could be anything really...

Comment (ibrasho) [inline src/Illuminate/Routing/Route.php]: You can store anything is the `$action` array...

File: src/Illuminate/Routing/Route.php [hunk 4]
```diff
@@ -905,13 +905,14 @@ public function getActionName()
     }
 
     /**
-     * Get the action array for the route.
+     * Get the action array or specific key for the route.
      *
+     * @param  string  $key
      * @return array
```
Comment (ibrasho) [inline src/Illuminate/Routing/Route.php]: Should we settle for `array|string`  @GrahamCampbell ?

File: src/Illuminate/Routing/Route.php (hunk 3 above)
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: `array|mixed` is wrong. It would have to be just `mixed`.

File: src/Illuminate/Routing/Route.php (hunk 4 above)
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: I think `array_string` is fine. :)

Comment (GrahamCampbell) [inline src/Illuminate/Routing/Route.php]: I think `array|string` is fine. :)

**Comment by taylorotwell** [comment]
Just get the full action and then access the key you want.

**Pull request closed without merging**